pkg/smi/client.go

```diff
@@ -280,6 +291,10 @@ func (c *client) ListTrafficTargets(options ...TrafficTargetListOption) []*smiAc
 			continue
 		}
 
+		if !isValidTrafficTarget(trafficTarget) {
+			continue
```

Comment:
Would it make sense to `log.Trace().Msgf()` the fact that we found an invalid traffic target?

Reply:
Should we have a prometheus counter for invalid TrafficTargets? That might be a really good counter to add.
I took the liberty to create the following issue for this #4187